### PR TITLE
JAVA-859: Faster start when there are no nodes for a DC in RF

### PR DESCRIPTION
**Pull request by clohfink**

### CODE REVIEW

**Comment by tolbertam** [comment]
looks good to me!  :+1:

File: driver-core/src/test/java/com/datastax/driver/core/NetworkTopologyStrategyTest.java [hunk 1]
```diff
@@ -571,6 +585,23 @@ public void networkTopologyStrategyExampleTopologyTest() {
     }
 
     @Test(groups = "unit")
+    public void networkTopologyStrategyNoNodesInDCTest() {
+        long t1 = System.currentTimeMillis();
+        Map<Token, Set<Host>> replicaMap =  networkTopologyStrategy(rf(DC1, 2), rf(DC2, 2))
+                .computeTokenToReplicaMap(largeRingTokenToPrimary, largeRing);
+        assertThat((System.currentTimeMillis() - t1) < 45000);
```
Comment (olim7t) [inline driver-core/src/test/java/com/datastax/driver/core/NetworkTopologyStrategyTest.java]: Should be:

```
assertThat(System.currentTimeMillis() - t1).isLessThan(45000);
```

Also, maybe we could lower the limit a bit. On my laptop, the optimized version consistently runs under a second.

**Comment by olim7t** [comment]
Could you reference the ticket in `changelog/README.md`?

**Comment by olim7t** [comment]
Excellent! Last request before I merge: could you squash your two commits and rebase on the current 2.0? It's OK to force push after that.

**Comment by tolbertam** [comment]
Excellent, looks good to me! :+1:

**Pull request merged**